scripts/build_vocab_mapping.py

```diff
@@ -67,27 +67,25 @@ def parse_args():
 def main():
     args = parse_args()
 
-    if not os.path.exists(args.token_freq_path):
-        raise FileNotFoundError(
-            f"Token frequency file not found: {args.token_freq_path}"
-        )
+    token_freq_path = Path(args.token_freq_path)
+    if not token_freq_path.exists():
+        raise FileNotFoundError(f"Token frequency file not found: {token_freq_path}")
 
-    token_freq_dict = torch.load(args.token_freq_path, weights_only=False)
+    token_freq_dict = torch.load(token_freq_path, weights_only=True)
 
     d2t, t2d = build_vocab_mappings_from_distribution(
         token_freq_dict=token_freq_dict,
         draft_vocab_size=args.draft_vocab_size,
         target_vocab_size=args.target_vocab_size,
     )
 
-    output_dir = os.path.dirname(args.output_path)
-    if output_dir:
-        os.makedirs(output_dir, exist_ok=True)
+    output_path = Path(args.output_path)
+    output_dir = output_path.parent
```

Comment:
This line is the cause of the [issue mentioned above ](https://github.com/vllm-project/speculators/pull/157/files/bfa577015859eeb3d9463a545028e892ca438d70..6a48b9f73140ea03fbd79a1cc260456dc2dc07fd#r2519543737). We should just use `output_path` directly.